### ISSUE TITLE
Dynamic refinementLists

### ISSUE THREAD

**Issue by dianos** [algolia/dynamic-faceting-guide#20]
Hi @Haroenv and @julienpa
I've tried implementing your examples of dynamic facets. However I have hundreds of different attribute facets, a few per each item type. Instead of importing each attribute as separate facet (which according to algolia's Documentation is not advisable). Is there a way to have refinementlists working from a set of hierarchical facets? So passing in a string like so:
Color > Black
Color > White
Battery Capacity > 1670
Battery Capacity > 4000
Battery Type > LiIon
Battery Type > Lipoly
Would generate multiple refinements lists that look something like:
<img width="189" alt="Disjointed_dynamic_refinmentlist" src="https://user-images.githubusercontent.com/4541410/128011409-cafcfb14-9a40-40f7-a25c-c4d95f4c4828.png">

**Comment by Haroenv** [comment]
It's not really possible due to the amount of queries that necessitates, but there's some workarounds: https://codesandbox.io/s/0ynjy0z530

**Comment by dianos** [comment]
Thanks @Haroenv that example no longer works with current instantsearch.js ;  Also it does not achieve the desired functionality: to filter results based on different attribute sets.  Instead it adds results when one filters in multiple sets.

---------------
What I'm asking is if it's possible to build on a recent example like that one you linked here:
https://preview.algolia.com/dynamic-faceting/
but instead of loading all the facets separately work of a set of generic facets.  Ideally the data in the facets could be like in the hierarchical facets.   
I don't see why this would necessitate a large amount of queries. In fact I already tried building this on top of 10 generic facets but the problem with doing it this way is: when you search across different item types you will get bleeding of possible values between these generic refinement lists.

**Comment by Haroenv** [comment]
>  when you search across different item types you will get bleeding of possible values between these generic refinement lists.

Exactly, the solution to this is doing a separate query per level, as it requires different filters to be applied

**Comment by dianos** [comment]
There is two levels.   One with headers and one with values (+headers potentially).   

If one were to generate refinementlists dynamically based on a lvl1 facet with headers.  Then filter results based on the facet that contains the values.  

So what you're saying is that it's just impossible to filter on the same facet in each of the refinmentlists?   So you would have to basically generate multiple facets with all the potential values and filter based on a differently named facet containing the same data, correct?

**Comment by Haroenv** [comment]
I haven't researched in this area in a while, but essentially you'll craft a request where there's a query for each one of the individual lists (a bit like how the hierarchical refinements does today, but then once per facet that's open, in your case every facet). To get the right counts you'll need to do an additional query for every refined value where you search with all filters applied except the facet value of that one.

```json
[
  // let's say color has a selected value
  { facets: ['chipset'], facetFilters: ['type:chipset', 'color:red'] },
  { facets: ['color'], facetFilters: ['type:color'] },
]
```

All in all, this experience doesn't really offer much over a list of refinementLists, I'm not sure why that doesn't fit for you.

**Comment by dianos** [comment]
>All in all, this experience doesn't really offer much over a list of refinementLists, I'm not sure why that doesn't fit for you.

I'm good with having a dynamic list of refinementLists.  The refinementLists could be collapsed initially that's not an issue.  I have hundreds different Item types.  With a few potentially different refinementLists per Item Type.   I can't really make a separate facet for every single attribute to pass to these refinementLists.   This is why the hierarchical structure would be ideal.  

The only (ugly) solution I have so far is:
Creating a few generic facets, each of them containing all the possible attributes for that item, and doing some display magic to only show the relevant values based on the heading of a given refinmentlist.

**Comment by dianos** [comment]
@Haroenv 
According to algolia's documentation we should keep the amount of searchable facets to a minimum:

>Reduce the number of attributes for faceting. We recommend keeping this list as small as possible, with only the strict minimum for your search UI.
https://www.algolia.com/doc/faq/index-configuration/my-facet-and-hit-counts-are-not-accurate/

If every item has say a maximum of 15 dynamic attributes;  but I have hundreds of different attribute sets, based on the type of item.  How would i go about implementing this efficiently? Note that new attributes can be added with data updates.

**Comment by Haroenv** [comment]
Sorry, I only work on the UI, I'm not sure what the best performance / backend considerations are @dianos, but you can contact support@algolia.com instead for more help!
